geograypher/meshes/meshes.py

```diff
@@ -162,19 +164,57 @@ def load_mesh(
             simplify_tol_meters=ROI_simplify_tol_meters,
         )
 
-        # Downsample mesh if needed
+        # Downsample mesh if needed transfer textures from original mesh to downsampled mesh
         if downsample_target != 1.0:
             # TODO try decimate_pro and compare quality and runtime
             # TODO see if there's a way to preserve the mesh colors
             # TODO also see this decimation algorithm: https://pyvista.github.io/fast-simplification/
             self.logger.info("Downsampling the mesh")
-            self.pyvista_mesh = self.pyvista_mesh.decimate(
+            intermediate_mesh = self.pyvista_mesh.decimate(
                 target_reduction=(1 - downsample_target)
             )
```

Comment:
A comment here would be helpful, and maybe rename intermediate mesh to something a little more descriptive.